base/compiler/ssair/inlining.jl

```diff
@@ -1077,6 +1077,20 @@ function ispuretopfunction(@nospecialize(f))
         istopfunction(f, :promote_type)
 end
 
+function intrinsic_tfunc_optimize(f::IntrinsicFunction, atypes::Vector{Any})
+    is_pure_intrinsic_optimize(f) || return nothing
+    args = atypes[2:end]
+    _all(@nospecialize(a) -> isa(a, Const), args) || return nothing
+    iidx = Int(reinterpret(Int32, f::IntrinsicFunction)) + 1
+    if iidx < 0 || iidx > length(T_IFUNC)
+        # invalid intrinsic
+        return nothing
+    end
+    (min, max, _) = T_IFUNC[iidx]
```

Comment:
Seems like you're trying to do error checking here, then only get around to doing half of it? I think we can reuse most of `pure_eval_call` here, rather than implementing an arbitrary subset of it.

Reply:
ah, oops that code got lost somewhere. I don't think `pure_eval_call` handles intrinsics though.

Reply:
huh, that seems sort of a little bit silly. I didn't realize we had a third copy of that code sitting around too.

Reply:
So should I add support for intrinsics to pure_eval_call then? I would kind of like to avoid the try/catch since we do have nothrow modeling for intrinsics.

Reply:
In any case, I've just added the error check here for now.

Reply:
Yeah, it's probably OK to reuse that support, although a bit less general (can't handle some functions where it might throw, but computing that set is pretty tedious—e.g. `sizeof_tfunc`). But it doesn't look like you've got any calls to the no-throw predicate right now.

Reply:
Seems to still be missing a call to the nothrow-test or a try/catch to handle it?